src/optimizer/query_rewrite.c

```diff
@@ -5495,27 +5495,21 @@ qo_rewrite_outerjoin (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *c
       prev_spec = NULL;
       for (spec = node->info.query.q.select.from; spec; prev_spec = spec, spec = spec->next)
 	{
-	  if (spec->info.spec.join_type == PT_JOIN_LEFT_OUTER
-	      || (spec->info.spec.join_type == PT_JOIN_RIGHT_OUTER && prev_spec))
+	  /* check outer join spec. */
```

Comment:
prev_spec is necessary ? it looks not useless.

Reply:
I removed prev_spec.
Using mq_is_outer_join_spec(), right outer join is checked more effectively.

Reply:
OK.